master/internal/db/postgres_cluster.go

```diff
@@ -9,9 +9,13 @@ import (
 )
 
 // GetOrCreateClusterID queries the master uuid in the database, adding one if it doesn't exist.
-func (db *PgDB) GetOrCreateClusterID() (string, error) {
-	newUUID := uuid.New().String()
-
+func (db *PgDB) GetOrCreateClusterID(telemetryID string) (string, error) {
+	var newUUID string
+	if telemetryID != "" {
+		newUUID = telemetryID
```

Comment:
maybe warn in logs if setting doesn't take (row already exists). and update the func comment to explain what passing "" does.